tests/e2e/smoke/gauntlet_test.go

```diff
@@ -0,0 +1,104 @@
+package smoke_test
+
+import (
+	"fmt"
+	"math/big"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/smartcontractkit/chainlink-terra/tests/e2e"
+	"github.com/smartcontractkit/chainlink-terra/tests/e2e/common"
+	"github.com/smartcontractkit/helmenv/environment"
+	"github.com/smartcontractkit/helmenv/tools"
+	"github.com/smartcontractkit/integrations-framework/actions"
+	"github.com/smartcontractkit/integrations-framework/client"
+	"github.com/smartcontractkit/integrations-framework/gauntlet"
+)
+
+var _ = Describe("Terra Gauntlet @gauntlet", func() {
+	var (
+		e     *environment.Environment
+		g     *gauntlet.Gauntlet
+		nodes []client.Chainlink
+		nets  *client.Networks
+		nkb   []common.NodeKeysBundle
+		err   error
+	)
+
+	BeforeEach(func() {
+		By("Deploying the environment", func() {
+			e, err = environment.DeployOrLoadEnvironment(
+				e2e.NewChainlinkTerraEnv(1),
+				tools.ChartsRoot,
+			)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = e.ConnectAll()
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		By("Setting up client", func() {
+			networkRegistry := client.NewNetworkRegistry()
+			networkRegistry.RegisterNetwork(
+				"terra",
+				e2e.ClientInitFunc(),
+				e2e.ClientURLSFunc(),
+			)
+			nets, err = networkRegistry.GetNetworks(e)
+			Expect(err).ShouldNot(HaveOccurred())
+			nodes, err = client.ConnectChainlinkNodes(e)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		By("Funding Wallets", func() {
+			_, nkb, err = common.DefaultOffChainConfigParamsFromNodes(nodes)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			err = common.FundOracles(nets.Default, nkb, big.NewFloat(5e12))
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		By("Setup Gauntlet", func() {
+			_, err := exec.LookPath("yarn")
+			Expect(err).ShouldNot(HaveOccurred())
+
+			cwd, _ := os.Getwd()
+			err = os.Chdir(filepath.Join(cwd + "../../../.."))
+			Expect(err).ShouldNot(HaveOccurred())
+
+			gauntletBin := fmt.Sprintf(
+				"%s%s",
+				filepath.Join(cwd, "../../../bin/gauntlet-terra-"),
+				gauntlet.GetOsVersion(),
+			)
+			g, err = gauntlet.NewGauntlet(gauntletBin)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			terraNodeUrl, err := e.Charts.Connections("localterra").LocalURLByPort("lcd", environment.HTTP)
+			Expect(err).ShouldNot(HaveOccurred())
+			g.NetworkConfig = common.GetDefaultGauntletConfig(terraNodeUrl)
+			g.WriteNetworkConfigMap()
+		})
+	})
+
+	Describe("Run Gauntlet Commands", func() {
+		It("should upload the contracts", func() {
+			_, err = g.ExecCommandWithRetries([]string{
+				"upload",
+				g.Flag("version", "local"),
```

Comment:
We should probably specify the network flag here. 
Actually, I think the Action failed, but gave a false positive cause this incorrect `process.exit` ? https://github.com/smartcontractkit/chainlink-terra/runs/5029805743?check_suite_focus=true#step:5:19

Reply:
The network is always applied: https://github.com/smartcontractkit/integrations-framework/blob/main/gauntlet/gauntlet.go#L69
with a unique one generated for each test when calling NewGauntlet: https://github.com/smartcontractkit/integrations-framework/blob/main/gauntlet/gauntlet.go#L46

As for that particular failure you linked to, my tests have nothing to do with that run although my gauntlet test did have a false positive and didn't run on pull request. I will make it run on pull request and see if I can figure out my own false positive. I will also add that false positive you pointed out to my todo list!

Reply:
You're right, I looked at the wrong test 😅